core/src/main/java/me/leoko/advancedban/utils/Command.java

```diff
@@ -542,7 +543,13 @@ else if(args.length == 2 && !args[0].matches("\\d+"))
 
     Command(String permission, String regex, TabCompleter tabCompleter, Consumer<CommandInput> commandHandler,
             String usagePath, String... names) {
-        this(permission, (args) -> String.join(" ", args).matches(regex), tabCompleter, commandHandler, usagePath, names);
+        final Pattern pattern = Pattern.compile(regex);
+        this.permission = permission;
+        this.syntaxValidator = (args) -> pattern.matcher(String.join(" ", args)).matches();
+        this.tabCompleter = tabCompleter;
+        this.commandHandler = commandHandler;
+        this.usagePath = usagePath;
+        this.names = names;
```

Comment:
I think it's better to call the other constructor here (as it was before) to prevent code duplication

Reply:
I changed the code to use Matcher#matches instead of String#matches because the regexes are pretty big and imo is better to have them compiled.